### PR TITLE
cherry pick [CDAP-20804] into release 6.9

### PR DESCRIPTION
**Pull request by seanfeld**
[CDAP-20804] : Fix tink directive add for replication transformations

# cherry pick 20804 into release 6.9
## Description
Summary of changes

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [x] Cherry Pick

[CDAP-20804]: https://cdap.atlassian.net/browse/CDAP-20804?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/e2e-test/features/wrangler.spanner.feature [hunk 1]
```diff
@@ -54,6 +54,7 @@ Feature: Wrangler Spanner Tests
     Then Verify URL navigation
 
   @WRANGLER_SPANNER_TEST
+  @ignore
```
Comment (njbriggs) [inline src/e2e-test/features/wrangler.spanner.feature:57]: Is this test breaking in this branch?

Comment (seanfeld) [reply src/e2e-test/features/wrangler.spanner.feature:57]: @njbriggs i figured it was a flaky test everywhere? @sumengwang ?

also we have this failure Non-resolvable parent POM for io.cdap.cdap:cdap-ui:6.9.3: Could not find artifact io.cdap.cdap:cdap:pom:6.9.3 in sonatype (https://oss.sonatype.org/content/groups/public) and 'parent.relativePath' points at wrong local POM @ line 23, column 11 -> [Help 2]

Comment (njbriggs) [reply src/e2e-test/features/wrangler.spanner.feature:57]: Are we missing a `-SNAPSHOT` somewhere?

Comment (sumengwang) [reply src/e2e-test/features/wrangler.spanner.feature:57]: I believe 6.9.3 https://github.com/cdapio/cdap-build/releases/tag/v6.9.3 was created after this test triggering. I'm rerunning it

Comment (sumengwang) [reply src/e2e-test/features/wrangler.spanner.feature:57]: why are you disabling the spanner test

Comment (seanfeld) [reply src/e2e-test/features/wrangler.spanner.feature:57]: that test is super flaky in dev @sumengwang why would it be less flaky here?

Comment (sumengwang) [reply src/e2e-test/features/wrangler.spanner.feature:57]: ok, didn't see it's already ignored in develop

File: pom.xml [hunk 2]
```diff
@@ -23,7 +23,7 @@
   <parent>
     <groupId>io.cdap.cdap</groupId>
     <artifactId>cdap</artifactId>
-    <version>6.9.3</version>
+    <version>6.9.4</version>
```
Comment (sumengwang) [inline pom.xml]: -SNAPSHOT

**Review by rmstar — APPROVED** [review]

**Pull request merged**